compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -606,71 +783,124 @@ MMAScheduleAttr::getContractionLayout(vector::ContractionOp contractOp) const {
   MMAAttr::SingleSubgroupLayout cOrders =
       mmaAttr.getCSingleSubgroupLayoutOrder();
 
-  SmallVector<int64_t, 2> cThreadBasis = cCounts.thread;
-  SmallVector<int64_t, 2> cDataDuplicate = mmaAttr.getCDataDuplicate();
-  for (auto [idx, duplicateFactor] : llvm::enumerate(cDataDuplicate)) {
-    cThreadBasis[idx] *= duplicateFactor;
+  auto [m, n] = opInfo.getResultFullMNIndex();
+  int64_t cRank = opInfo.getCRank();
+
+  // Get the M and N dims w.r.t. the dimensions of the C matrix. cMDims and
+  // cNDims are the M and N dimensions of the C matrix in the order they are
+  // iterated over in the contraction.
+  SmallVector<int64_t> cMDims = opInfo.outMDims;
+  SmallVector<int64_t> cNDims = opInfo.outNDims;
+  SmallVector<int64_t> cBatchSizes(cRank, 1);
+  SmallVector<int64_t> cSubgroupSizes(cRank, 1);
+  SmallVector<int64_t> cOverallOrder(cRank, 0);
+  for (auto [i, dim] : llvm::enumerate(cMDims)) {
+    cBatchSizes[dim] = batchMSizes[i];
+    cSubgroupSizes[dim] = subgroupMBasis[i];
+    cOverallOrder[dim] = mDimVec[i];
+  }
+  for (auto [i, dim] : llvm::enumerate(cNDims)) {
+    cBatchSizes[dim] = batchNSizes[i];
+    cSubgroupSizes[dim] = subgroupNBasis[i];
+    cOverallOrder[dim] = nDimVec[i];
   }
-  applyPermutationToVector(cThreadBasis, cOrders.thread);
+
+  // Dummy 1 for the k dimension.
+  subgroupBasis.push_back(1);
+
+  SmallVector<bool> cActiveSubgroups(cRank + 1, true);
+  cActiveSubgroups.back() = false;
 
   auto cLayout = permuteAndCreateNestedLayout(
```

Comment:
I wonder if we can create a layout and then permute it using the transpose functions exposed from the interface? Might be much easier to do.